engine.py

```diff
@@ -85,10 +85,25 @@ def get_bishop_moves(self, r, c, moves):
 
 			return parameter(s):
 			None
- 		"""
+		"""
 
- 		##TODO
- 		pass
+		##TODO
+		if self.light_to_move:
+			direction = [(-1,-1), (1,1),(1,-1), (-1,1)] # possible direction
+			for d in direction:
+				for i in range(1,8):
+					rownum = r + d[0] * i
+					colnum = c + d[1] * i
+					if 0 <= rownum <8 and 0 <= colnum < 8: # making sure r and c on board
```

Comment:
Use parenthesis to isolate different parts of a boolean expression

if (0 <= rownum < 8) and (0 <= colnum < 8)

And use len(self.board) instead of 8 for scalability